sympy/integrals/risch.py

```diff
@@ -711,6 +756,20 @@ def decrement_level(self):
         self.case = self.cases[self.level]
         return None
 
+def rewrite_tans(n, t):
+    from sympy.functions.combinatorial.numbers import nC
+    from sympy.core.compatibility import as_int
+
+    s0 = 0
+    n = as_int(n)
+    for k in range(1, (n + 1)//2 + 1):
+        s0 += nC(n, 2*k-1)*t**(2*k - 1)*(-1)**(k + 1)
```

Comment:
It is probably intentional, but I see no reason why it could not accept Integers.
There is `range(n + 1)` in the code, but n could easily be converted to int.